contracts/TaskBook.sol

```diff
@@ -0,0 +1,89 @@
+pragma solidity ^0.4.4;
+
+import './AccountManager.sol';
+
+contract TaskBook is AccountManager {
+
+	uint private numTasks = 0;
+	uint constant maxSolvers = 10;
+	uint constant maxChallengers = 10;
+
+	event NewTask(uint taskID, uint minDeposit, uint blockNumber);
+	event SolveTask(uint indexed taskID, address solver, bytes32 taskData, uint minDeposit);
+	event SubmitSolution(uint taskID, bytes32 solution, uint minDeposit, bytes32 taskData);
+
+	mapping(uint => Task) private tasks;
+	mapping(uint => bytes32) private solverRandomBits;
+	mapping(uint => mapping(address => uint)) private challengers;
+
+	struct Task {
+		address owner;
+		address[] solvers;
+		address selectedSolver;
+		uint minDeposit;
+		bytes32 taskData;
+		uint numSolvers;
+		address[] challengers;
+		uint numChallengers;
+	}
+
+	//Task Issuers create tasks to be solved
+	function newTask(uint minDeposit, bytes32 taskData, uint numBlocks) returns (bool) {
+		require(balances[msg.sender] >= minDeposit);
+		tasks[numTasks] = Task(msg.sender, new address[](maxSolvers), 0x0, minDeposit, taskData, 0, new address[](maxSolvers), 0);
```

Comment:
You probably meant
```
tasks[numTasks] = Task(msg.sender, new address[](maxSolvers), 0x0, minDeposit, taskData, 0, new address[](maxChallengers), 0);
```
I don't know if we can have max values for those, though.

Reply:
Oh yeah, good catch. I was wondering about the max values as well. I found that leaving it dynamic is more expensive and can cause out of gas error with not that much code. So put a cap on it for time being.

Reply:
Ok. Your arrays have dynamic size, you just initialize them to be large enough. The problem is iterating over the arrays, this cannot really be done if they can grow arbitrarily.